apps/GROMACS/ARCHER2_2023.4_cpe15.md

```diff
@@ -0,0 +1,305 @@
+Instructions for compiling GROMACS 2023.4 for ARCHER2 using CPE 15 compilers
+============================================================================
+
+These instructions are for compiling GROMACS 2023.4 on [ARCHER2](https://www.archer2.ac.uk) using the CPE 15 compilers.
+
+Download and Unpack the GROMACS source code
+-------------------------------------------
+
+Download and unpack the source
+
+```bash
+wget http://ftp.gromacs.org/pub/gromacs/gromacs-2023.4.tar.gz
+tar -xvf gromacs-2023.4.tar.gz
+cd gromacs-2023.4
+export SRC=$(pwd)
+```
+
+Setup correct modules and build environment
+-------------------------------------------
+
+Setup the programming environment:
+
+```bash
+module load cpe/22.12
+module load cray-python/3.9.13.1
+module load cray-fftw/3.3.10.3
+module load cmake/3.21.3
+export LD_LIBRARY_PATH=$CRAY_LD_LIBRARY_PATH:$LD_LIBRARY_PATH
+```
+
+These options are needed to be set correctly for any of the following steps.
+
+Configure and build the parallel, single-precision build
+--------------------------------------------------------
+
+Create a build directory in the source tree
+
+```bash
+cd ${SRC}
+mkdir build_mpi
+cd build_mpi
+```
+
+Use CMake to configure the build and then build and install. Remember to set the install 
+prefix to somewhere you have permission to write to.
+
+```bash
+export CC=cc
+export CXX=CC
+export GMX_INSTALL=/work/y07/shared/apps/core/gromacs/2023.4
+
+cmake                                       \
+      -DGMX_MPI=ON                          \
+      -DGMX_OPENMP=ON                       \
+      -DGMX_GPU=OFF                         \
+      -DGMX_DOUBLE=OFF                      \
+      -DGMX_BUILD_OWN_FFTW=ON               \
```

Comment:
This line would make GROMACS ignore earlier loaded `cray-fftw`. That's not incorrect (one can build own FFTW or use Cray's, either works), but first loading the module and then ignoring it seems a bit strange.

Reply:
Hi Andrey,

Thank you for taking the time to comment. I've had a look in our module, and it turns out that even though I was building the GROMACS FFTW routines, the cray ones were being picked up:

```
~> module load gromacs/2024.2
~> ldd /work/y07/shared/apps/core/gromacs/2024.2/bin/gmx_mpi | grep fftw
        libfftw3f_omp.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3f_omp.so.mpi31.3 (0x00007f780946c000)
        libfftw3_omp.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3_omp.so.mpi31.3 (0x00007f7809265000)
        libfftw3f.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3f.so.mpi31.3 (0x00007f780023e000)
        libfftw3.so.mpi31.3 => /opt/cray/pe/fftw/3.3.10.3/x86_rome/lib/libfftw3.so.mpi31.3 (0x00007f77ffd37000)
```

I'll update the documentation, removing the GROMACS-FFTW routine compilation.